dagger.nim

```diff
@@ -0,0 +1,22 @@
+import pkg/chronos
+
+type
+  DaggerPeer* = ref object
+
+proc newDaggerPeer*: DaggerPeer =
+  DaggerPeer()
+
+proc listen*(peer: DaggerPeer, address: TransportAddress) =
```

Comment:
We should use the same or as close as possible of an API to existing IPFS implementations - https://github.com/ipfs/js-ipfs/tree/master/docs/core-api.

Reply:
I'll address this in a later commit.

Reply:
Should be resolved with the latest couple of commits